transit_odp/avl/post_publishing_checks/weekly/tests/test_summary.py

```diff
@@ -442,3 +442,74 @@ def test_ppc_summary(
     # 2 x 25 = 50 for all fields matching vehicles
     assert df.iloc[1]["Successful match with TXC"] == 50
     assert df.iloc[1]["%match"] == "50%"
+
+
+@patch("json.load")
+def test_ppc_summary_no_data(
+    json_mock: Mock,
+    ppc_summary_data: PostPublishingChecksSummaryData,
+    daily_report_mock: PostPublishingCheckReport,
+) -> None:
+    daily_report_mock.vehicle_activities_completely_matching = 0
+    daily_report_mock.vehicle_activities_analysed = 0
+    data = [
+        {
+            "AVLtoTimetable matching summary": [
+                {
+                    "SIRI field": "LineRef",
+                    "TXC match field": "LineName",
+                    "Total vehicleActivities analysed": 0,
+                    "Total count of SIRI fields populated": 0,
+                    "%populated": "0.0%",
+                    "Successful match with TXC": "0",
+                    "%match": "0.0%",
+                    "Notes": "dummy_1",
+                },
+            ]
+        },
+        {
+            "AVLtoTimetable matching summary": [
+                {
+                    "SIRI field": "LineRef",
+                    "TXC match field": "LineName",
+                    "Total vehicleActivities analysed": 0,
```

Comment:
do one of these need to be an empty string to trigger an NA?

Reply:
If the total vehicles analysed is 0, then you get a divide-by-zero in the calculation of the two percentage figures, which fills in `nan`.

Reply:
Well in that case, take my approval sir!